Feature/splitimage: Add tool to split large images (like DnD tactical maps) into ready-to-print multipage PDFs

DemetriSam opened the pull request:
Implements a new SplitImage tool that divides a large image into multiple PDF pages matching the selected paper format (A4 or Letter or whatever that works for you) and padding.
Supports unit selection (pt, mm, in) and scaling options based on grid squares (pixels per square, number of squares, and units per square).
Intended for preparing printable DnD maps or other large-format images.

DemetriSam committed:
refactor: extract zip logic and deps into ./utils

DemetriSam committed:
feat: add split tool divides images into PDF pages ready for print

oasis9 commented:
Wondering if there's any progress on this? Just came looking for the same function to complement the images to PDF function.

DemetriSam commented:
> Wondering if there's any progress on this? Just came looking for the same function to complement the images to PDF function.

The function is implemented in my fork.

So far you can try to deploy it from the fork on my github page or I can send you the link to the deployed project.

Chesterkxng commented:
Hey, sorry for the delay! It will be reviewed soon, if possible by next weekend.

oasis9 commented:
Glad to hear from you both 😊 

@DemetriSam Appreciate your efforts. I think I'll deploy myself, I like the peace of mind of being able to run this offline, thanks for the offer though.

I've achieved what I needed to for the time being but this will really help with future workflows where I need to split an image into equal size rows (and/or columns) and then make a PDF out of them. This is a really cool project, thanks to each of you for your time and efforts.

Chesterkxng committed:
chore:: loading text added to locales

Chesterkxng committed:
chore:
- computing function moved to service.ts
- loading state added to resultcomponent

Chesterkxng committed:
chore:
- new splitImagesToPdf function
- functions optimized

Chesterkxng committed:
Merge branch 'main' into pr/DemetriSam/274

Chesterkxng committed:
chore: zip util file deleted

Chesterkxng committed:
chore:
- unused keys removal
- pdf extension specified

Chesterkxng committed:
chore: updated meta

Chesterkxng reviewed (approved):
Good work, thank you !

Chesterkxng commented:
Hey @DemetriSam @oasis9, the feature is now live! Thanks for your contribution and your patience.

DemetriSam commented:
@Chesterkxng glad to hear it, thank you! I appreciate the opportunity to contribute to your project
